plugins/woocommerce/includes/wc-template-functions.php

```diff
@@ -3108,15 +3108,14 @@ function woocommerce_form_field( $key, $args, $value = null ) {
 
 				break;
 			case 'select':
-				$field   = '';
 				$options = '';
 
 				if ( ! empty( $args['options'] ) ) {
 					foreach ( $args['options'] as $option_key => $option_text ) {
 						if ( '' === $option_key ) {
-							// If we have a blank option, select2 needs a placeholder.
+							// A blank option is the proper way to set a placeholder. If one is supplied we make sure the placeholder key is set for selectWoo.
 							if ( empty( $args['placeholder'] ) ) {
-								$args['placeholder'] = $option_text ? $option_text : __( 'Choose an option', 'woocommerce' );
+								$args['placeholder'] = $option_text ? : __( 'Choose an option', 'woocommerce' );
```

Comment:
This feels weird to me. What is the value if `$option_text` is set? `undefined`? `null`? Maybe it's worth being a bit more explicit here with an actual value for the `?` part of the ternary expression

Reply:
The [shorthand ternary operator](https://stitcher.io/blog/shorthand-comparisons-in-php#shorthand-ternary-operator) in the following line looks broken:
```php
$args['placeholder'] = $option_text ? : __( 'Choose an option', 'woocommerce' );
```

Shouldn't it be `?:` instead of `? :`? Note the extra space between the the chars.

Reply:
It's not broken, space is optional and formatter added it. But I'll revert this because linting doesn't allow shorthand ternaries.

@alexflorisca it's the shorthand operator, in the cases the value itself decides if it stays the same or gets assigned a new one, enabling us to ommit that part. It was equivalent to the unchanged line, but the linter doesn't like it so I'll remove.